### PR TITLE
enable storing extends/implements in TypesIntoElements#store()

### PR DESCRIPTION
**Pull request by CharlesZ-Chen**
This PR resolved the set different sizes issues posted in pascaliUWat/CFI:
- https://github.com/pascaliUWat/checker-framework-inference/issues/15
- https://github.com/pascaliUWat/checker-framework-inference/issues/14

Also fixed bugs caused by enabling storing extends/implements for a class element :

1. add if-check in storeClassExtends(): do nothing when the passed class symbol is an interface or java.lang.Object. This fixed a Nullpointer Exception. 

2. changed `typeOffset` in `SuperTypeAppiler#annotateSupers()` to always be -1. detail reasons are commented in code

The original review of this code is here: 

https://github.com/opprop/checker-framework/pull/5

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java [hunk 1]
```diff
@@ -81,7 +81,16 @@ public int getElementIndex() {
      */
     @Override
     public int getTypeCompoundIndex(Attribute.TypeCompound anno) {
-        return anno.getPosition().type_index;
+        int type_index = anno.getPosition().type_index;
+        // TODO: this is a temporary workaround of a bug in jsr308-langtools
```
Comment (smillst) [inline framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java]: Is this a jsr308-langtools bug or a general langtools bug?  Has a bug report been summited?

Comment (CharlesZ-Chen) [inline framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java]: I'm not sure whether this is a jsr308 specific bug or a general one. The buggy code is located at com.sun.tools.javac.jvm.ClassReader in jsr308-langtools:
https://bitbucket.org/typetools/jsr308-langtools/src/71bc2f5e1d950f803437ae4c7c93ca22846b6fd8/src/share/classes/com/sun/tools/javac/jvm/ClassReader.java?at=default&fileviewer=file-view-default#ClassReader.java-1592

@wmdietl Prof. Dietl has reported this mismatch to Oracle a few weeks ago.

Comment (smillst) [inline framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java]: Yes, it's just a langtools bug.  Here's the OpenJDK bug report: https://bugs.openjdk.java.net/browse/JDK-8164519.  Could you please change `jsr308-langtools` to just `langtools` on this line and add the link to the bug report?

Comment (CharlesZ-Chen) [inline framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java]: Already changed and added the bug report link.

**Comment by smillst** [comment]
Looks good.  Thanks!

**Comment by smillst** [comment]
FYI, I had to revert this pull request because it caused failures in daikon-typecheck.  (See #880 for description)

**Comment by smillst** [comment]
I also added the failing test case here: checker/jtreg/nullness/annotationsOnExtends/.

**Comment by CharlesZ-Chen** [comment]
Thanks for the test case! Will have look on it.

**Pull request merged**